provider/kubernetes/kubernetes.go

```diff
@@ -155,6 +167,50 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 	return nil
 }
 
+func (p *Provider) updateIngressStatuses(k8sClient Client) error {
+	//Only run these ingress updates if config is provided. This will reduce overhead
+	if p.IngressEndpoint.Hostname != "" || p.IngressEndpoint.IP != "" || p.IngressEndpoint.PublishedService != "" {
+		ingresses := k8sClient.GetIngresses()
+
+		for _, i := range ingresses {
+			annotationIngressClass := getAnnotationName(i.Annotations, annotationKubernetesIngressClass)
+			ingressClass := i.Annotations[annotationIngressClass]
+
+			if !p.shouldProcessIngress(ingressClass) {
+				continue
+			}
+
+			if p.IngressEndpoint.PublishedService != "" {
+				s := strings.Split(p.IngressEndpoint.PublishedService, "/")
+				serviceNamespace, serviceName := s[0], s[1]
+				svc, exists, _ := k8sClient.GetService(serviceNamespace, serviceName)
+				//Set the status to the status of the fronting LB service
+				if exists && svc.Status.LoadBalancer.Ingress == nil {
+					//service exists, but has no LoadBalancer status
+					log.Debugf("Skipping updating Ingress %s/%s due to service %s having no status set", i.Namespace, i.Name, p.IngressEndpoint.PublishedService)
+					continue
+				}
+				if exists {
+					err := k8sClient.UpdateIngressStatus(i.Namespace, i.Name, svc.Status.LoadBalancer.Ingress[0].IP, svc.Status.LoadBalancer.Ingress[0].Hostname)
+					if err != nil {
+						log.Errorf("Cannot update Ingress %s/%s due to error: %s", i.Namespace, i.Name, err)
```

Comment:
I think the log can be transform to an error.

Reply:
If someone submits a non-compliant ingress, we just want to log an error, and continue processing. We don't want to interrupt the processing of other ingresses.